### ISSUE TITLE
No version manager name reported in CLI output, consider adding it

### ISSUE THREAD

**Issue by olivierlacan** [asdf-vm/asdf#1467]
### Is your feature request related to a problem? Please describe

Hi there, excuse the snarky issue title, but it would be nice if asdf introduced itself in directories where it can't find `.tool-versions` instead of printing a cryptic message (which appears partially broken) without attribution. 

Otherwise this output message can be quite confusing: 

```
$ npm outdated
No version is set for command npm
Consider adding one of the following versions in your config file at
nodejs 16.13.2
```

Clearly there's some missing text here, probably assuming a location to a `.tool-versions` could be interpolated at the end of the second to last line? 

Formatting-wise, showing `nodejs 16.13.2` without any list formatting is an odd choice as well, since at different viewport width it appears to be the end of the sentence above.

### Describe the proposed solution

Change this message to this instead: 

```
$ npm outdated
asdf found no installed version for command `npm`.
You can add a version in `.tool-versions`, the following version is available: nodejs 16.13.2
```

That's of course assuming there's no alternative configuration file, but that's nothing a good link to a documentation page on the website won't fix if needed.

### Describe similar `asdf` features and why they are not sufficient

Since asdf is intercepting this call to `npm` I don't think this relates to other asdf features.

### Describe other workarounds you've considered

The only workaround would be for someone to be aware of `which npm` or `which -a npm` which sadly may be a bit too much to ask in some cases.

In my case, I forgot I installed `asdf` while also using `volta` and a Homebrew-installed version of node. I know, I know, that's a bit silly but I don't think it's that rare considering [this StackOverflow thread](https://stackoverflow.com/questions/74935373/no-version-is-set-for-command-npm-tool-versions-file-missing)

**Comment by hyperupcall** [comment]
Yeah, I agree that it's pretty confusing. I've done some preliminary work in https://github.com/hyperupcall/asdf/commit/3a6d35b263de09aa5db2df5d119b8e4ed39f4f46 to address this

Before I make a PR, I'm still trying to figure out if prepending the terminal messages with `asdf: ` should be a thing enabled by default, or behind an environment variable. Sort of leaning towards the former, with a possible opt-out.

**Comment by olivierlacan** [comment]
@hyperupcall I agree, disabling `asdf: <command>` prepending should definitely be an option, maybe that's something that could be enabled for any existing installs vs. net new ones to avoid breaking any stdout capture folks might have? 

Any way to reliably detect that? Maybe empty install directories?

**Comment by hyperupcall** [comment]
@olivierlacan Since the changes only prepending things that are printed to standard output, my vote is to just roll out these changes. The contents of the actual error message aren't being changed, so stuff like `grep -q 'unknown shim'` will still work. In my opinion, if they do something like `grep -Eq '$unknown shim'` then the "breaking change" is on them. I'm just a regular contributor, so maybe an asdf core maintainer will disagree.

I'm not really a fan of testing for empty install directories since things can be uninstalled and that behavior change will violate the principle of least astonishment. If there really needs to be some dynamic opt-in mechanism, I'm thinking of making a tweak of the shim file:

```sh
#!/usr/bin/env bash
# asdf-plugin: nodejs 19.4.0
ASDF_NEW_PREFIXED_PRINTS=yes exec /.../git/asdf/bin/asdf exec "npm" "$@" # asdf_allow: ' asdf '
```

So things work given the following routine:

```sh
display() {
  if [ "$ASDF_NEW_PREFIXED_PRINTS" = yes ]; then
    printf 'asdf: '
  fi

  # shellcheck disable=SC2059
  printf "$@"
}
```

A side effect of this, is that under normal use, people won't see a change to the output, only when `exec`ing into other executables. I don't really like this either, since is also a divergence in behavior, which is why I'm leaning towards just making a clean breaking change

I suppose an environment variable can be added, but in my opinion, that just increases the maintenance burden, for a change that should really be a clean fix of the old behavior.

related: #1162

**Comment by hyperupcall** [comment]
The PR that fixes this implements something similar to the above comment - but messages are _always_ printed when `asdf exec` is ran.

This is an improvement over my original shim-approach, since shims need to be either manually updated or they are only automatically updated when installing a new shim. The new approach works consistently.
